flow/flow.go

```diff
@@ -277,6 +277,47 @@ func (p FlowProtocol) MarshalJSON() ([]byte, error) {
 	return []byte("\"" + p.String() + "\""), nil
 }
 
+// MarshalJSON serialize a TransportLayer in JSON
+func (tl *TransportLayer) MarshalJSON() ([]byte, error) {
```

Comment:
I'm not sure that the marshal and the unmarshal functions are needed, I submitted a PR to remove them
#1124

Reply:
I'll fix this in my PR, let's merge it